src/main/java/graphql/validation/Validator.java

```diff
@@ -157,6 +160,15 @@ public List<AbstractRule> createRules(ValidationContext validationContext, Valid
         UniqueObjectFieldName uniqueObjectFieldName = new UniqueObjectFieldName(validationContext, validationErrorCollector);
         rules.add(uniqueObjectFieldName);
 
+        DeferDirectiveLabel deferDirectiveLabel = new DeferDirectiveLabel(validationContext, validationErrorCollector);
+        rules.add(deferDirectiveLabel);
+
+        DeferDirectiveOnRootLevel deferDirectiveOnRootLevel = new DeferDirectiveOnRootLevel(validationContext, validationErrorCollector);
+        rules.add(deferDirectiveOnRootLevel);
+
+        DeferDirectiveOnValidOperation deferDirectiveOnValidOperation = new DeferDirectiveOnValidOperation(validationContext, validationErrorCollector);
+        rules.add(deferDirectiveOnValidOperation);
+
```

Comment:
I think these rules should be controlled by the same graphql context control that controls execution

they should be added sure BUT inside them they should only act IF the flag is on

Reply:
Good point 👍

Reply:
Added on each validation rule a check if ExperimentalApi.ENABLE_INCREMENTAL_SUPPORT is true